routing/router.go

```diff
@@ -1832,7 +1834,30 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 					paySession.ReportEdgeFailure(
 						failedEdge,
 					)
+					return
+				}
+
+				// Update routing hints.
+				var direction lnwire.ChanUpdateFlag
+				if failedEdge.direction == 1 {
+					direction = lnwire.ChanUpdateDirection
+				}
+
+				fChanID := failedEdge.channelID
+				uBaseFee := lnwire.MilliSatoshi(update.BaseFee)
+				uFeeRate := lnwire.MilliSatoshi(update.FeeRate)
+				uMinHTLC := lnwire.MilliSatoshi(
+					update.HtlcMinimumMsat)
```

Comment:
This vars are introduced because of line length I assume. But maybe that indicates that now is the time to extract this closure to a separate function.

Reply:
There are plenty of internal functions that are called only once, it does help to make the code more understandable. Inline functions can be useful especially if a lot of local vars are bound.

How about just creating a function taking `failedEdge` and `update` and returning the `ChannelEdgePolicy`? Or better, updating the signature of `UpdateEdgePolicy` to take  `failedEdge` and `update`. Also consistent with other pay session funcs that take an `edgeLocator` too.